pkgs/build-support/cc-wrapper/default.nix

```diff
@@ -299,6 +299,12 @@ stdenv.mkDerivation {
       hardening_unsupported_flags+=" stackprotector fortify pie pic"
     ''
 
+    + optionalString targetPlatform.isWasm ''
+      hardening_unsupported_flags+=" stackprotector fortify pie pic"
+    '' + optionalString (targetPlatform.isWasm && libc != null) ''
+      echo "--allow-undefined-file=${libc}/share/wasm32-wasi/undefined-symbols.txt" >> $out/nix-support/cc-ldflags
```

Comment:
IIRC, lld will look for files named `foo.imports` in the same directories that it will look for libraries in. We should simply give this file the `.imports` extension and keep it in a `lib` directory.

Reply:
Yeah that sounds good! Maybe a symlink just to keep the structure from wasi-sysroot.

Reply:
@matthewbauer I wouldn't be surprised if wasi-sysroot creates the required file already.